### PR TITLE
Nonblocking ap

### PR DESCRIPTION
**Pull request by DrBoolean**
Appears to function, but needs some refactor love.

### CODE REVIEW

**Comment by rjmk** [comment]
Excited for this when it lands :heart_eyes:

File: lib/task.js [hunk 1]
```diff
@@ -128,12 +128,55 @@ Task.prototype.chain = function _chain(f) {
...
+
+    return allState = [thisState, thatState];
+
+    function guardFn(f, isFail) {
+      return function(x) {
+        if(isFail) return f(x);
+
+        if (val) {
+          delayed(function(){ cleanupBoth(allState) });
+          return f(func(x));
```
Comment (rjmk) [inline lib/task.js]: Shouldn't this be `f(x(val))`?

Comment (safareli) [inline lib/task.js]: @rjmk you are right. if `guardVal` is called, and some truthy value is stored in `val` and then `guardFn` is called `func` will be undefined.
also if `that` task will resolve to undefined or some falsy value and then `this` is resolved, `val` will be falsy so no one will call `func`

**Comment by DrBoolean** [comment]
closing in favor of #33

**Pull request closed without merging**